torch/_prims/__init__.py

```diff
@@ -2724,7 +2743,11 @@ def _gather_aten(tensor: torch.Tensor, dims: List[int], indices: List[torch.Tens
     if len(indices) == 1:
         for i in range(indices[0].ndim):
             r = r.movedim(dims[0] + i, i)
-    return r.contiguous()  # TODO: don't contiguous here
+    # TODO: returning contiguous here is necessary for the overall
+    # implementation of __getitem__ to return the correct strides,
+    # but it means the implementation of gather here is suboptimal;
+    # we're gathering into the wrong contiguity.
+    return r.contiguous()
```

Comment:
So it turns out making this contiguous is correct, and the old implementation gives strides that are inconsistent with PyTorch's native implementation!